src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+from typing import Optional, Union
+import numpy as np
+import networkx as ntx
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import PathPatch
+from matplotlib.path import Path
+
+
+class SchedulingProblem:
+    def __init__(self,
+                 num_agents: int = 3,
+                 num_tasks: int = 3,
+                 sat_cutoff: Union[float, int] = 0.99,
+                 seed: int = 42):
+        """Schedule `num_tasks` tasks among `num_agents` agents such that
+        every agent performs exactly one task and every task gets assigned
+        exactly one agent.
+
+        Parameters
+        ----------
+        num_agents (int) : number of agents available to perform all tasks.
+        Default is arbitrarily chosen as 3.
+
+        num_tasks (int) : number of tasks to be performed. Default is
+        arbitrarily chosen as 3.
+
+        sat_cutoff (float or int) : If float, this is the satisfiability
+        cut-off calculated as the ratio between the number of tasks that get
+        assigned an agent to the total number of tasks. A fraction between 0
+        and 1. If int, this is the target cost for the underlying QUBO
+        solver. Default is 0.99 (i.e., 99% of the total number of tasks get
+        assigned an agent).
+
+        seed (int) : Seed for PRNG used in problem generation.
+        """
+        self._num_agents = num_agents
+        self._agent_ids = range(num_agents)
+        self._agent_attrs = None
+        self._num_tasks = num_tasks
+        self._task_ids = range(num_tasks)
+        self._task_attrs = None
+        self._sat_cutoff = sat_cutoff
+        self.graph = None
+        self.adjacency = None
+        self._random_seed = seed
+
+    @property
```

Comment:
Beautiful code, following all suggestions I know of. Complete docstrings, properties, getter/setter...
Thanks for keeping the code quality high!

Reply:
Only minor point of criticism: You may want to add input validation.

Reply:
Do you think it is reasonable that we will ever do it? If so, please add an issue to Github. If not, just ignore.